cfme/tests/cloud_infra_common/test_tag_objects.py

```diff
@@ -14,14 +14,12 @@
 from cfme.infrastructure.provider import InfraProvider
 from cfme.infrastructure.virtual_machines import Vm, Template
 from cfme.web_ui import Quadicon, mixins, toolbar as tb
-from utils import providers
 from utils.appliance.implementations.ui import navigate_to
 
 
 @pytest.fixture(scope="module")
-def setup_first_provider():
-    providers.setup_a_provider_by_class(InfraProvider)
-    providers.setup_a_provider_by_class(CloudProvider)
+def cloud_and_infra_provider(cloud_provider, infra_provider):
```

Comment:
ie if we have one module with `infra_provider` and another that gets executed right after with `cloud_provider` you have exactly the same situation unless we are removing providers when going from one test module to another.